.gitlab-ci.yml

```diff
@@ -86,6 +88,12 @@ default:
     - /^pre-v[0-9]+\.[0-9]+-[0-9a-f]+$/
     - web
 
+.chaos-only:                       &chaos-only
+  only:
+    variables:
+      - $CI_COMMIT_MESSAGE =~ /\[chaos:(basic|medium|large)\]/  # i.e add [chaos:basic] in commit message to trigger
```

Comment:
This does not add up with `&build-only` in the `build-linux-substrate:` job. It uses just the latest `only:`.

Reply:
> > HarryHong force-pushed the harry/chaostest-init branch 4 times, most recently from [d075f78](https://github.com/paritytech/substrate/commit/d075f789aad3e4e32cd43f4beee917a2fb94ada3) to [b7eb4d5](https://github.com/paritytech/substrate/commit/b7eb4d513a7ce301d0913b595a03dcfb20a910db) 9 days ago
> 
> @HarryHong please don't force-push to a branch that already received a review. Pushing individual commits instead makes reviewing a lot easier as one can easily understand what changed since the last review.
> 
> All commits are squashed into a single one before merging anyways ;)

What do you suggest here to do when a conflict with master happened?

Reply:
> This does not add up with `&build-only` in the `build-linux-substrate:` job. It uses just the latest `only:`.

As far as i tested, i think it works fine. Would this cause other issues?

Reply:
> What do you suggest here to do when a conflict with master happened?

Resolve the conflict and commit it.

Reply:
you are breaking `build-linux-substrate:` and `build-linux-subkey: ` jobs making them work only by this your rule.

Reply:
https://gitlab.parity.io/parity/substrate/-/pipelines/99070
I added a test condition tag to ```build-linux-substrate:```, seems like it works no problem.
 I could not find anywhere stating stacked ```only``` will only work for last one. 

ex: 
```
build-linux-substrate:             &build-binary
  stage:                           build
  <<:                              *collect-artifacts
  <<:                              *docker-env
  <<:                              *build-only
  <<:                              *chaos-only
  <<:                              *test-only
...
.test-only:                       &test-only
  only:
    variables:
      - $CI_COMMIT_MESSAGE =~ /\[chaos:test\]/ 
```
In example: I add [chaos:basic] to commit-message, it still proc the ```*chaos-only``` condition.

Reply:
commented above